### DISCUSSION TITLE
Most efficient way to pre-process data in all patches of a spool?

### DISCUSSION THREAD

**Discussion by ahmadtourei** [DASDAE/dascore#216] [Q&A] [answered]
What would be the most efficient way to pre-process data in all patches of a spool (e.g., if I want to scale all data by a scaling factor)? Then, I want to have a spool containing the same patches but with updated data.  Currently, I'm using a for loop to pre-process the data:

```python
import dascore as dc 

sp = dc.spool(data_path)
patch_0 = sp[0]

gauge_length = 15.9524
scale = float((116/gauge_length)*1e-9)

p_list_data_scaled = []

for i, patch in enumerate (sp):
    print ("working on patch ", i)
    data_scaled = sp[i].data * scale
    a = sp[i].new(data=data_scaled)
    p_list_data_scaled.append(a)

new_sp_data_scaled = dc.spool(p_list_data_scaled)
```
But this way, first, I'm using a for lop to do this (not the most efficient way), and second, I need to create a MEmoerySpool from the new patches. 

Thanks!

**Comment by d-chambers** [comment, marked as answer]
That's probably the best way until #200 gets implemented, provided your data are small enough to fit into the MemorySpool. Otherwise you may need to dump each patch to disk after processing then create a new DirectorySoool
